api/scripts/database/drop-database.js

```diff
@@ -11,9 +11,24 @@ const DB_TO_DELETE_NAME = url.pathname.slice(1);
 
 url.pathname = '/postgres';
 
+const databaseInstanceCanBeDropped = () => {
+  if (process.env.NODE_ENV === 'test') {
+    return true;
+  }
+  if (!process.env.ALLOW_DATABASE_DROP || process.env.ALLOW_DATABASE_DROP !== 'true') {
+    return false;
+  }
+  return true;
+};
+
+if (!databaseInstanceCanBeDropped()) {
+  logger.info(`Database will not be dropped (check environment variables NODE_ENV and ALLOW_DATABASE_DROP)`);
+  process.exit(0);
```

Comment:
Par curiosité, si on ne peut pas drop est-ce que c'est une bonne idée de continuer ici ? Qu'est-ce qui se passerait si on faisait comme ci-dessous ?
```suggestion
  logger.error(`Database will not be dropped (check environment variables NODE_ENV and ALLOW_DATABASE_DROP)`);
  process.exit(1);
```

Reply:
La réponse est dans la description de PR (spoiler: la CI et les tests E2E ne passent plus)

Reply:
Ok, j'avais compris que pour ces envs on allait activer ALLOW_DATABASE_DROP et que donc on ne passerait pas dans ce "if", qui n'est là que comme garde fou. Si ce n'est pas le cas, est-ce qu'on a besoin de cette variable d'env ALLOW_DATABASE_DROP ? Dans quels cas sera-t-elle activée ?

Reply:
Elle ne sera alimentée qu'en local, sur le poste du développeur - c'est pour cela qu'elle est décommentée dans le`sample.env`

Reply:
Je peux mettre à jour la description de PR si ce n'est pas clair

Reply:
Ok, je vois le truc un peu différemment maintenant :

- en local, on veut faire db:delete + db:create + db:migrate + db:seed
- sur une RA ou en CI, on veut faire db:create + db:migrate + db:seed (alors c'est pas vraiment un reset ?)

Si c'est bien ça, pourquoi appeler le même db:reset pour les deux ? Sur RA c'est plus un init & seed ? Si j'ai bien compris, le db:reset ne fera pas le db:delete en RA ou CI de toute façon, donc il n'a de reset que le nom ?

Reply:
Suite à discussion avec @jonathanperret , on peut avoir tous les bienfaits sans les inconvénients en ne supprimant la BDD que si l'on est sur Scalingo, ce que l'on peut savoir avec les variable d'environnement  https://doc.scalingo.com/platform/app/environment#runtime-environment-variables

Reply:
_sauf si l'on est sur Scalingo_ plutôt…

Reply:
Je vais fermer cette PR qui ne ressemble plus à grand chose et en ouvrir 2 autres:
- une qui empêche un problème déjà présent: suppression BDD sur Scalingo
- une qui déconnecte les utilisateurs avant de supprimer par LBDD